github/actions/client.go

```diff
@@ -62,23 +63,66 @@ type Client struct {
 	ActionsServiceAdminTokenExpiresAt *time.Time
 	ActionsServiceURL                 *string
 
-	RetryMax     *int
-	RetryWaitMax *time.Duration
+	retryMax     int
+	retryWaitMax time.Duration
 
 	creds           *ActionsAuth
 	githubConfigURL string
 	logger          logr.Logger
 	userAgent       string
 }
 
-func NewClient(ctx context.Context, githubConfigURL string, creds *ActionsAuth, userAgent string, logger logr.Logger) (ActionsService, error) {
+type ClientOption func(*Client)
+
+func WithUserAgent(userAgent string) ClientOption {
+	return func(c *Client) {
+		c.userAgent = userAgent
+	}
+}
+
+func WithLogger(logger logr.Logger) ClientOption {
+	return func(c *Client) {
+		c.logger = logger
+	}
+}
+
+func WithRetryMax(retryMax int) ClientOption {
+	return func(c *Client) {
+		c.retryMax = retryMax
+	}
+}
+
+func WithRetryWaitMax(retryWaitMax time.Duration) ClientOption {
+	return func(c *Client) {
+		c.retryWaitMax = retryWaitMax
+	}
+}
+
+func NewClient(ctx context.Context, githubConfigURL string, creds *ActionsAuth, options ...ClientOption) (ActionsService, error) {
 	ac := &Client{
 		creds:           creds,
 		githubConfigURL: githubConfigURL,
-		logger:          logger,
-		userAgent:       userAgent,
+		logger:          logr.Discard(),
+
+		// retryablehttp defaults
+		retryMax:     4,
+		retryWaitMax: 30 * time.Second,
+	}
+
+	for _, option := range options {
+		option(ac)
 	}
 
+	retryClient := retryablehttp.NewClient()
+
+	// TODO: this silences retryclient default logger, do we want to provide one
```

Comment:
It would be great if we could selectively enable the logging, for example, let retryablehttp log requests only when the log level is DEBUG or finer.
That way I'd be able to set the log level to DEBUG when I'm curious why job run delays are unusually long and the retrylablehttp logs would reveal that it's due to excessive retries caused by e.g. an actions service incident.

Reply:
Anyway, we'd better address that in another pull request, if necessary, to avoid scope creeping this one!

Reply:
I think we might want to have some logging on error cases, especially to log out some HTTP headers so we can check telemetry on the GitHub service.
Ex: `x-github-activity-id`, `ActivityId`, etc.

Reply:
again, it doesn't have to be this PR. 😄